xrootd/xrootd_config_test.go

```diff
@@ -18,17 +18,19 @@
  *
  ***************************************************************/
 
-package main
+package xrootd
 
 import (
 	"testing"
 
+	"github.com/spf13/viper"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-func TestCreateXrootdConfig(t *testing.T) {
-	configPath, err := configXrootd()
+func TestXrootDConfig(t *testing.T) {
+	viper.Reset()
```

Comment:
I don't understand how this works.  I would have expected something like this to be needed:

```
	dirname := t.TempDir()
	os.Setenv("PELICAN_XROOTD_RUNLOCATION", dirname)  # Or equivalent viper.Set call
	defer os.Unsetenv("PELICAN_XROOTD_RUNLOCATION")
	config_dirname := t.TempDir()
	viper.Reset()
	require.NoErr(t, err)
```

(and then the rest of the test)

Reply:
The issue is that before, the xrootdrunlocation was being set to a temporary directory by the scitokensconfig test that then gets destroyed, so when it tries to create the file, it can't because the directory doesn't exist. Currently the directory is unset, so it's "", which does exist. Which I realize definitely needs to be fixed in the way you are suggesting, but that's why it doesn't fail, because it can find the directory and create the file.